src/lib/arch/riscv/Instruction.cc

```diff
@@ -0,0 +1,196 @@
+#include "simeng/arch/riscv/Instruction.hh"
+
+#include <algorithm>
+#include <cassert>
+#include <vector>
+
+#include "InstructionMetadata.hh"
+
+namespace simeng {
+namespace arch {
+namespace riscv {
+
+const Register Instruction::ZERO_REGISTER = {RegisterType::GENERAL, 0};
+
+Instruction::Instruction(const Architecture& architecture,
+                         const InstructionMetadata& metadata)
+    : architecture_(architecture), metadata(metadata) {
+  decode();
+}
+
+Instruction::Instruction(const Architecture& architecture,
+                         const InstructionMetadata& metadata, uint8_t latency,
+                         uint8_t stallCycles)
+    : architecture_(architecture), metadata(metadata) {
+  latency_ = latency;
+  stallCycles_ = stallCycles;
+
+  decode();
+}
+
+Instruction::Instruction(const Architecture& architecture,
+                         const InstructionMetadata& metadata,
+                         InstructionException exception)
+    : architecture_(architecture), metadata(metadata) {
+  exception_ = exception;
+  exceptionEncountered_ = true;
+}
+
+InstructionException Instruction::getException() const { return exception_; }
+
+void Instruction::setSourceRegisters(const std::vector<Register>& registers) {
+  assert(registers.size() <= MAX_SOURCE_REGISTERS &&
+         "Exceeded maximum source registers for a RISCV instruction");
+
+  sourceRegisterCount = registers.size();
+  operandsPending = registers.size();
+
+  for (size_t i = 0; i < registers.size(); i++) {
+    auto reg = registers[i];
+    if (reg == Instruction::ZERO_REGISTER) {
+      // Any zero-register references should be marked as ready, and
+      //  the corresponding operand value zeroed
+      operands[i] = RegisterValue(0, 8);
+      operandsPending--;
+    }
+    sourceRegisters[i] = reg;
+  }
+}
+void Instruction::setDestinationRegisters(
+    const std::vector<Register>& registers) {
+  assert(registers.size() <= MAX_DESTINATION_REGISTERS &&
+         "Exceeded maximum destination registers for a RISCV instruction");
+  destinationRegisterCount = registers.size();
+  std::copy(registers.begin(), registers.end(), destinationRegisters.begin());
+}
```

Comment:
These functions are un-used and can be removed

Reply:
Removed